graphql_compiler/compiler/emit_sql.py

```diff
@@ -103,6 +108,23 @@ def _find_folded_outputs(ir):
     return folded_outputs
 
 
+def _find_x_count_filters(ir):
+    """For each fold path, find all filters involving _x_count."""
```

Comment:
Isn't it simpler to find just the set of fold scope locations that use `_x_count` in any way (filter or output it)? It seems to me like that's how this result is being used, in a convoluted way. Also it seems like `_get_fold_outputs` will be simplified. Correct me if I'm missing something.